src/storage/query/GetDstBySrcProcessor.cpp

```diff
@@ -168,7 +182,19 @@ folly::Future<std::pair<nebula::cpp2::ErrorCode, PartitionID>> GetDstBySrcProces
                         }
                       }
                       return std::make_pair(nebula::cpp2::ErrorCode::SUCCEEDED, partId);
-                    });
+                    })
+      .thenError(folly::tag_t<std::bad_alloc>{},
```

Comment:
Since we allow to throw exception here, we need to handle it because storage expect no exception. For example, in `GetDstBySrcProcessor::runInMultipleThread`:

```
folly::collectAll(futures).via(executor_).thenTry([this](auto&& t) mutable {
    CHECK(!t.hasException());
    ...
```

Besides, if we catch the exception in `runInMultipleThread`, we probably don't need the `onError` in `src/storage/BaseProcessor.h`

Reply:
from what I got from debugging, runInMultipleThread is running in another executor,  it seems cannot be catched outside, after added this onError, outside then can get exception.

Reply:
I'm curious about how we survive from crashing when `query_concurrently=true`. Because if we throw in `runInExecutor`, it will be checked against if the future has an exception in `runInMultipleThread`, just as the code in above comments. Do you have any screenshots? I saw you have on when `query_concurrently = false` in previous comments.

> there isn't short circuit in `collectAll`, but `collect` has. That's why I think we may encounter crash.

Reply:
folly::collectAll(futures).via(executor_).thenTry([this](auto&& t) mutable {
    CHECK(!t.hasException());
    const auto& tries = t.value();
    size_t sum = 0;
    for (size_t j = 0; j < tries.size(); j++) {
      if (tries[j].hasException()) {
        onError();
        return;
      }
      sum += results_[j].size();
    }

Reply:
![Screenshot from 2022-12-26 16-33-35](https://user-images.githubusercontent.com/1256658/209525715-a1cdc8d3-aa10-4d46-8a97-da1f0f3ffbf9.png)
![Screenshot from 2022-12-26 16-32-51](https://user-images.githubusercontent.com/1256658/209525725-1fad6451-aa39-496f-aa49-e7553e03ffd7.png)
@critical27 check this out

Reply:
My bad, I didn't see you already handle it around line 139.